pkg/api/validate/dynamic/dynamic.go

```diff
@@ -388,6 +392,29 @@ func (dv *dynamic) ValidateSubnets(ctx context.Context, oc *api.OpenShiftCluster
 	return nil
 }
 
+func (dv *dynamic) ValidateWorkerSku(ctx context.Context, location string, subscriptionID string, oc *api.OpenShiftCluster) error {
+	msiAuthorizer, err := dv.env.NewMSIAuthorizer(env.MSIContextRP, dv.env.Environment().ResourceManagerEndpoint)
+	if err != nil {
+		return err
+	}
```

Comment:
Why do we use `msiAuthorizer` here?

Reply:
I used `msiAuthorizer` because of this documentation https://github.com/Azure/ARO-RP/blob/master/docs/authorizers.md#msiauthorizer

> Used for bootstrapping: finding the CosmosDB key, populating the SKU list.